pkg/kubelet/cm/memorymanager/policy_static_test.go

```diff
@@ -1561,9 +1566,10 @@ func TestStaticPolicyAllocate(t *testing.T) {
 					v1.ResourceMemory: 512 * mb,
 				},
 			},
-			pod:           getPod("pod2", "container2", requirementsGuaranteed),
-			expectedError: fmt.Errorf("[memorymanager] failed to find the extended preferred hint"),
-			topologyHint:  &topologymanager.TopologyHint{NUMANodeAffinity: newNUMAAffinity(1), Preferred: true},
+			pod:               getPod("pod2", "container2", requirementsGuaranteed),
+			expectedError:     fmt.Errorf("[memorymanager] failed to find the extended preferred hint"),
+			shouldUnwrapError: true,
```

Comment:
is this really necessary? I don't understand when you would ever *not* unwrap it.

Meaning, can't you just do the following:
```
			err = p.Allocate(s, testCase.pod, &testCase.pod.Spec.Containers[0])
			if err ! nil {
				err = errors.Unwrap(err)
			}
```

Reply:
nope, this is is my attempt to avoid to always unwrap the error, but if we prefer to do the unwrapping unconditionally, I'm all for it - I think in some cases is the most sane solution. Fixing.